src/core/client/admin/components/UserRole/SiteModeratorModalSiteField.tsx

```diff
@@ -31,41 +31,42 @@ const SiteModeratorModalSiteField: FunctionComponent<Props> = ({
   disableLoadMore,
   loading,
 }) => {
-  const { input } = useField<string[]>("siteIDs");
-  const onChange = useCallback(
-    (siteID: string, selectedIndex: number) => () => {
-      const changed = [...input.value];
-      if (selectedIndex >= 0) {
-        changed.splice(selectedIndex, 1);
-      } else {
-        changed.push(siteID);
-      }
-
-      input.onChange(changed);
-    },
-    [input]
-  );
+  const [field] = useField("siteIDs");
 
   return (
     <FieldSet>
       <FormField>
         <Localized id="community-siteModeratorModal-selectSites">
           <Label>Select sites to moderate</Label>
         </Localized>
+        <pre>{field.value}</pre>
         <ListGroup className={styles.listGroup}>
-          {sites.map((site) => {
-            const selectedIndex = input.value.indexOf(site.id);
-            return (
-              <ListGroupRow key={site.id}>
-                <CheckBox
-                  checked={selectedIndex >= 0}
-                  onChange={onChange(site.id, selectedIndex)}
-                >
-                  {site.name}
-                </CheckBox>
-              </ListGroupRow>
-            );
-          })}
+          <FieldArray name="siteIDs">
+            {({ push, remove }) => (
```

Comment:
The `FieldArray` component provides array mutation methods that should allow us to build forms like our slack config form without the issues we've seen with sorting and correct indices. Unfortunately there is not a complementary `useFieldArray` hook though we could look into writing one.